apps/frontend/src/libs/components/icon/types/icon.type.ts

```diff
@@ -0,0 +1,9 @@
+import { type IconName } from "./icon-name.type.js";
+
+type Icon = {
+	className?: string | undefined;
+	name: IconName;
+	onClick?: () => void;
```

Comment:
icon shouldn't have className or onClick. If you need to modify the width or something else, pass a specific property. We can add onClick only to interactive elements like buttons, so if you need this, you need to create a separate component like IconButton.   

```suggestion
	name: IconName;
```